app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,53 @@
+<div class="container recipe-details">
+  <h2><%= @recipe.name %></h2>
+
+  <div class="row recipe-details-box1">
+    <div class="col-md-6">
+      <p><strong>Preparation Time:</strong> <%= @recipe.preparation_time %> minutes</p>
+    </div>
+
+    <div class="col-md-6 text-right">
+      <%= form_for @recipe, url: recipe_path(@recipe), method: :patch do |f| %>
+        <div class="form-group">
+          <%= f.label :public, class: 'mr-2' %>
+          <%= f.check_box :public, id: 'recipe_public', class: 'form-check-input' %>
+          <%= f.submit 'Toggle', class: 'btn btn-primary' %>
+        </div>
+      <% end %>
+    </div>
+  </div>
+
+  <p><strong>Cooking Time:</strong> <%= @recipe.cooking_time %> minutes</p>
+  <p><strong>Description:</strong> <%= @recipe.description %></p>
+  
+  <div class="recipe-details-btns mt-3">
+    <button class="btn btn-success">Generate Shopping List</button>
+    <% if current_user == @recipe.user %>
+      <%= link_to "Add Ingredient", new_recipe_recipe_food_path(recipe_id: @recipe.id), class: 'btn btn-primary' %>
+    <% end %>
```

Comment:
- [ ] Please make sure to finish the implementation of the "Generate Shopping List" button as it should be able to generate a shopping list that matches the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/general_shopping_list.png?raw=true) and URL `/general_shopping_list`
- [ ] Note for reviewers: Since I wasn't able to create a food item it wasn't possible to check whether the add ingredient functionality worked as expected so kindly check this for them as per the guideline.

Reply:
Concerning this->**Please make sure to finish the implementation of the "Generate Shopping List" button as it should be able to generate a shopping list that matches the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/general_shopping_list.png?raw=true) and URL /general_shopping_list**

 That feature is only for 3 group people and we are only 2 and that is not in our projcet requirements. So we just linked to to the shopping lists url
 
 Thank you

Reply:
**It's important to ensure that you create both success and failure paths for the update and destroy actions in your controllers. These paths play a crucial role in handling different scenarios and providing appropriate feedback to users. By implementing these paths, you can enhance the user experience and ensure that your application gracefully handles both successful and unsuccessful interactions.**

Resolved

Reply:
@zuhebahmed88091 My apologies there was an ambiguous requirement in the review guideline concerning generating a shopping list. You're right it should only be implemented for a group of 3 members. Thank you for bringing it to my attention!